jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/extensions/compress/CompressExtension.java

```diff
@@ -199,11 +204,14 @@ protected void decompress(ByteAccumulator accumulator, ByteBuffer buf) throws Da
             }
 
             int read;
-            while ((read = inflater.inflate(output)) >= 0)
+            
+            while (true)
             {
-                if (read == 0)
+                byte[] output = accumulator.newByteArray(DECOMPRESS_BUF_SIZE);
+                read = inflater.inflate(output);
+                if (read <= 0)
                 {
-                    if (LOG.isDebugEnabled())
+                    if (read == 0 && LOG.isDebugEnabled())
```

Comment:
Don't change the logic this much in the decompress method.
Focus on the ByteAccumulator only for this PR.